packages/flutter/lib/src/material/material_state.dart

```diff
@@ -630,96 +388,18 @@ class _MaterialStateUnderlineInputBorder extends MaterialStateUnderlineInputBord
-  /// at build time.
-  T resolve(Set<MaterialState> states);
-
-  /// Resolves the value for the given set of states if `value` is a
-  /// [MaterialStateProperty], otherwise returns the value itself.
-  ///
-  /// This is useful for widgets that have parameters which can optionally be a
-  /// [MaterialStateProperty]. For example, [InkWell.mouseCursor] can be a
-  /// [MouseCursor] or a [MaterialStateProperty<MouseCursor>].
-  static T resolveAs<T>(T value, Set<MaterialState> states) {
-    if (value is MaterialStateProperty<T>) {
-      final MaterialStateProperty<T> property = value;
-      return property.resolve(states);
-    }
-    return value;
-  }
-
-  /// Convenience method for creating a [MaterialStateProperty] from a
-  /// [MaterialPropertyResolver] function alone.
-  static MaterialStateProperty<T> resolveWith<T>(MaterialPropertyResolver<T> callback) => _MaterialStatePropertyWith<T>(callback);
-
-  /// Convenience method for creating a [MaterialStateProperty] that resolves
-  /// to a single value for all states.
-  ///
-  /// If you need a const value, use [MaterialStatePropertyAll] directly.
-  ///
-  // TODO(darrenaustin): Deprecate this when we have the ability to create
-  // a dart fix that will replace this with MaterialStatePropertyAll:
-  // https://github.com/dart-lang/sdk/issues/49056.
-  static MaterialStateProperty<T> all<T>(T value) => MaterialStatePropertyAll<T>(value);
-
-  /// Linearly interpolate between two [MaterialStateProperty]s.
-  static MaterialStateProperty<T?>? lerp<T>(
-    MaterialStateProperty<T>? a,
-    MaterialStateProperty<T>? b,
-    double t,
-    T? Function(T?, T?, double) lerpFunction,
-  ) {
-    // Avoid creating a _LerpProperties object for a common case.
-    if (a == null && b == null) {
-      return null;
-    }
-    return _LerpProperties<T>(a, b, t, lerpFunction);
-  }
-}
-
-class _LerpProperties<T> implements MaterialStateProperty<T?> {
-  const _LerpProperties(this.a, this.b, this.t, this.lerpFunction);
-
-  final MaterialStateProperty<T>? a;
-  final MaterialStateProperty<T>? b;
-  final double t;
-  final T? Function(T?, T?, double) lerpFunction;
-
-  @override
-  T? resolve(Set<MaterialState> states) {
-    final T? resolvedA = a?.resolve(states);
-    final T? resolvedB = b?.resolve(states);
-    return lerpFunction(resolvedA, resolvedB, t);
-  }
-}
-
-class _MaterialStatePropertyWith<T> implements MaterialStateProperty<T> {
-  _MaterialStatePropertyWith(this._resolve);
-
-  final MaterialPropertyResolver<T> _resolve;
-
-  @override
-  T resolve(Set<MaterialState> states) => _resolve(states);
-}
+typedef MaterialStateProperty<T> = WidgetStateProperty<T>;
 
 /// Convenience class for creating a [MaterialStateProperty] that
 /// resolves to the given value for all states.
-class MaterialStatePropertyAll<T> implements MaterialStateProperty<T> {
+class MaterialStatePropertyAll<T> extends WidgetStatePropertyAll<T>{
```

Comment:
It's because of the toString message saying specifically "MaterialStatePropertyAll". I was unsure if it was important if that backwards compatibility was preserved, so I left it as it's own thing.